src/currencymanager.cpp

```diff
@@ -0,0 +1,274 @@
+
+    You should have received a copy of the GNU General Public License
+    along with Acquisition.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <QWidget>
+#include <QtGui>
+#include <QDialog>
+#include <ctime>
+#include "QsLog.h"
+#include "application.h"
+#include "datamanager.h"
+#include "buyoutmanager.h"
+#include "currencymanager.h"
+#include "itemsmanager.h"
+#include "item.h"
+CurrencyManager::CurrencyManager(Application &app) : app_(app), data_(app.data_manager())
+{
+    if (data_.Get("base", "", "currency") == "")
+        InitCurrency();
+    LoadCurrency();
+    connect(&app_.items_manager(), SIGNAL(ItemsRefreshed(Items, std::vector<std::string>)),
+        this, SLOT(UpdateExaltedValue()));
+    connect(&app_.items_manager(), SIGNAL(ItemsRefreshed(Items, std::vector<std::string>)),
+        this, SLOT(SaveCurrencyValue()));
+}
+
+CurrencyManager::~CurrencyManager()
+{
+    SaveCurrencyBase();
+}
+
+void CurrencyManager::UpdateBaseValue(int ind, double value)
+{
+    currencys_[ind].base = value;
+}
+
+void CurrencyManager::UpdateExaltedValue()
+{
+    for( auto &currency : currencys_) {
+        if (currency.base != 0)
+            currency.exalt = currency.count / currency.base;
+    }
+}
+
+double CurrencyManager::TotalExaltedValue()
+{
+    double out = 0;
+    for (auto currency : currencys_) {
+        out += currency.exalt;
+    }
+    return out;
+}
+
+int CurrencyManager::TotalWisdomValue()
+{
+    int out = 0;
+    for (unsigned int i=0; i < wisdoms_.size(); i++) {
+        out += wisdoms_[i] * CurrencyWisdomValue[i];
+    }
+    return out;
+}
+
+void CurrencyManager::ClearCurrency()
+{
+    for(auto &currency : currencys_) {
+        currency.count = 0;
+    }
+    for( auto &wisdom : wisdoms_) {
+        wisdom = 0;
+    }
+}
+
+void CurrencyManager::InitCurrency()
+{
+    std::string value = "";
+    std::string header_csv = "";
+    for (auto curr : CurrencyAsString) {
+        value += "0;";
+        if (curr != "")
+            header_csv += curr + ";";
+    }
+    value.pop_back();//Remove the last ";"
+    header_csv += "Total value; Timestamp";
+    data_.Set("base", value, "currency");
+    data_.Set("last_value",value, "currency");
+    //Create header for .csv
+    QFile file("export_currency.csv");
+    if (file.open(QFile::WriteOnly | QFile::Truncate)){
+        QTextStream out(&file);
+        out  << header_csv.c_str();
+    }
+    else {
+        QLOG_WARN() << "CurrencyManager::ExportCurrency : couldn't open CSV export file ";
+    }
+}
+
+void CurrencyManager::LoadCurrency()
+{
+    //Way easier to use QT function instead of re-implementing a split() in C++
+    QStringList list = QString(data_.Get("base", "", "currency").c_str()).split(";");
+    for (int i=0; i < list.size(); i++) {
+        CurrencyItem curr;
+        curr.count = 0;
+        curr.name = CurrencyAsString[i];
+        //We can't use the toDouble function from QT, because it encodes a double with a ","
```

Comment:
Was this behavior actually observed? From my understanding and docs using `toDouble` should work just fine. We also set up the default locale to be C here https://github.com/xyzz/acquisition/blob/a2142fe8c933eb2ab51bf18ca1375e7f652d13c3/src/main.cpp#L50